data_diff/sqeleton/databases/snowflake.py

```diff
@@ -142,8 +142,27 @@ def _query(self, sql_code: Union[str, ThreadLocalInterpreter]):
         return self._query_conn(self._conn, sql_code)
 
     def select_table_schema(self, path: DbPath) -> str:
-        schema, name = self._normalize_table_path(path)
-        return super().select_table_schema((schema, name))
+        """Provide SQL for selecting the table schema as (name, type, date_prec, num_prec)"""
+        database, schema, name = self._normalize_table_path(path)
+
+        return (
+            "SELECT column_name, data_type, datetime_precision, numeric_precision, numeric_scale "
+            f"FROM {database + '.' if database else ''}information_schema.columns "
```

Comment:
Please extract the table path to `columns_schema_path` (or something like that) and use `'.'.join(columns_schema_path)`. I think it would be a bit cleaner.

I hope that was clear.

Reply:
Is this what you were thinking?
[2cea813](https://github.com/datafold/data-diff/pull/331/commits/2cea813e5e801cba2d965e6b25feb5556953e508)